src/Protocol/ChunkDataSerializer.cpp

```diff
@@ -502,7 +502,21 @@ inline void cChunkDataSerializer::Serialize477(const int a_ChunkX, const int a_C
 	// Write each chunk section...
 	ChunkDef_ForEachSection(a_BlockData, a_LightData,
 	{
-		m_Packet.WriteBEInt16(ChunkBlockData::SectionBlockCount);  // a temp fix to make sure sections don't disappear
+		// Count non-air blocks.
+		Int16 BlockCount = 0;
+		if (Blocks != nullptr)
+		{
+			for (size_t i = 0; i != ChunkBlockData::SectionBlockCount; i++)
+			{
+				const BLOCKTYPE BlockType = (*Blocks)[i];
+				if (BlockType != E_BLOCK_AIR)
+				{
+					BlockCount++;
+				}
```

Comment:
Brute forcing this hurts a bit. But I don't have any other ideas on how to do this faster

Reply:
Vectorise! Would be easy to do a vector comparison,mask and popcount but would add some more complexity to the code/builds. Or keep an extra account of the block count alongside the main data (definitely sounds like a source of bugs to me!).

Honestly this naive version probably wouldn't actually have much effect on perf anyway, this block should be memory-throughput bound streaming the data into the packet anyway and an extra iteration would probably get lost in the noise.

Reply:
I see, this is fine